File: ChallengeBackend/src/main/java/br/com/savedra/challengebackend/repository/PromotionRepository.java
package br.com.savedra.challengebackend.repository;

import br.com.savedra.challengebackend.model.Promotion;
import org.springframework.data.mongodb.repository.MongoRepository;
import java.util.List;

public interface PromotionRepository extends MongoRepository<Promotion, String> {
    List<Promotion> findBySegment(String segment);
}

File: ChallengeCRM/app/src/main/java/br/com/savedra/challengecrm/data/repository/PromotionRepository.kt
package br.com.savedra.challengecrm.data.repository

import br.com.savedra.challengecrm.data.api.PromotionApi
import br.com.savedra.challengecrm.model.Promotion

class PromotionRepository(private val promotionApi: PromotionApi) {

  suspend fun sendPromotion(promotion: Promotion) {
    promotionApi.createPromotion(promotion)
  }

  suspend fun getPromotions(userSegment: String? = null): List<Promotion> {
    val response = promotionApi.getPromotions(userSegment)
    return if (response.isSuccessful) {
        response.body() ?: emptyList()
    } else {
        emptyList()
    }
  }
}
